openlr/decoded_path.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "routing/road_graph.hpp"
+
+#include "base/exception.hpp"
+#include "base/newtype.hpp"
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "3party/pugixml/src/pugixml.hpp"
+
+class Index;
+
+namespace openlr
+{
+using Path = routing::RoadGraphBase::TEdgeVector;
+using routing::Edge;
+
+NEWTYPE(uint32_t, PartnerSegmentId);
+NEWTYPE_SIMPLE_OUTPUT(PartnerSegmentId);
+
+DECLARE_EXCEPTION(DecodedPathLoadError, RootException);
+DECLARE_EXCEPTION(DecodedPathSaveError, RootException);
+
+struct DecodedPath
+{
+  PartnerSegmentId m_segmentId;
+  Path m_path;
+};
+
+void WriteAsMappingForSpark(std::string const & fileName, std::vector<DecodedPath> const & paths);
+void WriteAsMappingForSpark(std::ostream & ost, std::vector<DecodedPath> const & paths);
+
+void PathFromXML(pugi::xml_node const & node, Index const & index, Path & path);
+void PathToXML(Path const & path, pugi::xml_node & node);
+}  // namespace openlr
+
+namespace routing
+{
+inline m2::PointD GetStart(Edge const & e) { return e.GetStartJunction().GetPoint(); }
```

Comment:
For example [here](https://github.com/mapsme/omim/pull/6778/files#diff-7b07224738848cbf38b274bb8b39283eR250).